numpy/random/src/philox/philox.h

```diff
@@ -32,6 +32,15 @@ static NPY_INLINE uint64_t mulhilo64(uint64_t a, uint64_t b, uint64_t *hip) {
   *hip = product >> 64;
   return (uint64_t)product;
 }
+#elif defined(_M_X64) || defined(_M_ARM64)
+#include <intrin.h>
+#pragma intrinsic(__umulh)
+static NPY_INLINE uint64_t mulhilo64(uint64_t a, uint64_t b, uint64_t *hip) {
+  uint64_t p0 = a * b;
+  uint64_t p1 = __umulh(a, b);
```

Comment:
Is this an intrnisic newly defined for arm64 + MSVC? Is there a parallel intrinsic for arm64 + gcc?

Reply:
Here is the general definition: https://docs.microsoft.com/en-us/cpp/intrinsics/umulh?view=msvc-160
And it is present as an arm64 intrinsic: https://docs.microsoft.com/en-us/cpp/intrinsics/arm64-intrinsics?view=msvc-160

I do not know if there is one for arm+gcc.

Reply:
Is this better on AMD64 than using `_umul128` directly?